Dockerfile

```diff
@@ -5,7 +5,7 @@ MAINTAINER Thomas VIAL
 RUN DEBIAN_FRONTEND=noninteractive apt-get update -q --fix-missing && \
 	apt-get -y upgrade && \
 	apt-get -y install --no-install-recommends \
-	postfix dovecot-core dovecot-imapd dovecot-pop3d dovecot-sieve dovecot-managesieved gamin amavisd-new spamassassin razor pyzor \
+	postfix dovecot-core dovecot-imapd dovecot-pop3d dovecot-sieve dovecot-managesieved gamin amavisd-new spamassassin razor pyzor libsasl2-modules \
```

Comment:
Unfortunately, yes. The config you have uses dovecot for SASL, but dovecot doesn't support client auth (http://www.postfix.org/SASL_README.html#client_sasl says "At this time, the Dovecot SASL implementation does not provide client functionality") and what is needed for AWS SES is for postfix to do SASL as a client. The same would be needed for pretty much any scenario where outgoing email gets relayed through an authenticated connection, unless I'm missing something...